linera-chain/src/block_tracker.rs

```diff
@@ -85,8 +101,79 @@ impl<'resources, 'blobs> BlockExecutionTracker<'resources, 'blobs> {
         })
     }
 
+    /// Executes a transaction in the context of the block.
+    pub async fn execute_transaction<C>(
+        &mut self,
+        transaction: Transaction<'_>,
+        round: Option<u32>,
```

Comment:
Maybe the round should also be a field in the `BlockExecutionTracker`. It's the same for every transaction.

(And possibly even the `&mut ExecutionStateView<C>`? Not sure.)

Reply:
Yes, I think it _could_ be moved there. I thought that MAYBE the same instance of `BlockExecutionTracker` can be applied on top of other things that change:
- round
- chain view

I.e. when the block is reproposed in a different round, on a different chain state, its transactions or replayed oracle responses don't change?